cmd/buildtools/utils.go

```diff
@@ -475,13 +491,17 @@ func MirrorChart(repo *repo.Entry, name, ver string) error {
 	}
 
 	dst := fmt.Sprintf("oci://%s", os.Getenv("CHARTS_DESTINATION"))
+	chartURL := fmt.Sprintf("%s/%s", dst, name)
 	logrus.Infof("verifying if destination tag already exists")
-	tmpf, err := hcli.Pull(dst, name, ver)
+	dstMeta, err := GetOCIChartMetadata(chartURL, name, ver)
 	if err != nil && !strings.HasSuffix(err.Error(), "not found") {
 		return fmt.Errorf("verify tag exists: %w", err)
 	} else if err == nil {
-		os.Remove(tmpf)
-		logrus.Warnf("cowardly refusing to override dst (tag %s already exist)", ver)
+		if srcMeta.AppVersion == dstMeta.AppVersion {
+			logrus.Infof("cowardly refusing to override dst (tag %s already exist)", ver)
+			return nil
+		}
+		logrus.Warnf("dst tag exists but app versions do not match (src: %s, dst: %s)", srcMeta.AppVersion, dstMeta.AppVersion)
```

Comment:
shouldn't the idea be to push in this case? why is there a return statement?

Reply:
doesnt that defeat the purpose of shrinkwrapping?